packages/@aws-cdk/lambda-layer-awscli/lib/awscli-layer.ts

```diff
@@ -1,19 +1,150 @@
+      const result = AwsCliLayer.requireWrapper(AwsCliLayer.assetPackageName, logs);
+      if (result) {
+        logs.push(`Successfully loaded ${AwsCliLayer.assetPackageName} from pre-installed packages.`);
+        return result;
+      }
+    }
+  }
+
+  /**
+   * @internal
+   */
+  public static _downloadPackage(targetVersion: string, logs: string[]): string | undefined {
+    const cdkHomeDir = cxapi.cdkHomeDir();
+    const downloadDir = path.join(cdkHomeDir, 'npm-cache');
+    const downloadPath = path.join(downloadDir, `${AwsCliLayer.assetPackageNpmTarPrefix}${targetVersion}.tgz`);
+
+    if (fs.existsSync(downloadPath)) {
+      logs.push(`Using package archive already available at location: ${downloadPath}`);
+      return downloadPath;
+    }
+    logs.push(`Downloading package using npm pack to: ${downloadDir}`);
+    childproc.execSync(`mkdir -p ${downloadDir}; cd ${downloadDir}; npm pack ${AwsCliLayer.assetPackageName}@${targetVersion} -q`);
+    if (fs.existsSync(downloadPath)) {
+      logs.push('Successfully downloaded using npm pack.');
+      return downloadPath;
+    }
+
+    logs.push('Failed to download using npm pack.');
+    return undefined;
+  }
+
+  private static readonly assetPackageName: string = '@aws-cdk/asset-awscli-v1';
+  private static readonly assetPackageNpmTarPrefix: string = 'aws-cdk-asset-awscli-v1-';
+
+  private static requireWrapper(id: string, logs: string[]): any {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-require-imports
+      const result = require(id);
+      if (result) {
+        logs.push(`require('${id}') succeeded.`);
+      }
+      return result;
+    } catch (e) {
+      logs.push(`require('${id}') failed.`);
+      const eAsError = e as Error;
+      if (eAsError?.stack) {
+        logs.push(eAsError.stack);
+      }
+    }
+  }
+
+  private static installAndLoadPackage(from: string, logs: string[]): any {
+    const installDir = AwsCliLayer.findInstallDir();
+    if (!installDir) {
+      logs.push('Unable to find an install directory. require.main.paths[0] is undefined.');
+      return;
+    }
+    logs.push(`Installing from: ${from} to: ${installDir}`);
+    childproc.execSync(`npm install ${from} --no-save --prefix ${installDir} -q`);
+    return AwsCliLayer.requireWrapper(path.join(installDir, 'node_modules', AwsCliLayer.assetPackageName, 'lib/index.js'), logs);
+  }
+
+  private static findInstallDir(): string | undefined {
+    if (!require.main?.paths) {
+      return undefined;
+    }
+    return require.main.paths[0];
+  }
+
   constructor(scope: Construct, id: string) {
-    super(scope, id, {
-      code: lambda.Code.fromAsset(path.join(__dirname, 'layer.zip'), {
+    const logs: string[] = [];
+    let fallback = false;
+
+    const targetVersion = AwsCliLayer.requireWrapper(path.join(__dirname, '../package.json'), logs).devDependencies[AwsCliLayer.assetPackageName];
```

Comment:
We can build the version in at build time, in the same script that copies the asset zip.

```
echo "export const LAYER_VERSION = '$(node -pe 'require("@aws-cdk/lambda-layer/package.json").version")';" > lib/layerversion.ts
```

Reply:
ooh good idea, updated.